pkg/apis/componentconfig/validation/validation.go

```diff
@@ -0,0 +1,109 @@
+	}
+	if 0 <= cc.HardPodAffinitySymmetricWeight && cc.HardPodAffinitySymmetricWeight >= 100 {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("hardPodAffinitySymmetricWeight"), cc.HardPodAffinitySymmetricWeight, "not in valid range 0-100"))
+	}
+	if cc.EnableContentionProfiling && !cc.EnableProfiling {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("enableContentionProfiling"), cc.EnableContentionProfiling, "can't set EnableContentionProfiling without first setting EnableProfiling"))
+	}
+	return allErrs
+}
+
+// ValidateClientConnectionConfiguration ensures validation of the ClientConnectionConfiguration struct
+func ValidateClientConnectionConfiguration(cc *componentconfig.ClientConnectionConfiguration, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if len(cc.KubeConfigFile) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("kubeConfigFile"), ""))
+	}
+	if cc.QPS < 0 {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("qps"), cc.QPS, "must be a non-negative value"))
+	}
+	return allErrs
+}
+
+// ValidateSchedulerAlgorithmSource ensures validation of the SchedulerAlgorithmSource struct
+func ValidateSchedulerAlgorithmSource(cc *componentconfig.SchedulerAlgorithmSource, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if cc.Policy == nil {
+		return allErrs
+	}
+	allErrs = append(allErrs, ValidateSchedulerPolicySource(cc.Policy, field.NewPath("schedulerPolicySource"))...)
+	return allErrs
+}
+
+// ValidateSchedulerPolicySource ensures validation of the SchedulerPolicySource struct
+func ValidateSchedulerPolicySource(cc *componentconfig.SchedulerPolicySource, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if cc.File == nil && cc.ConfigMap == nil {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("file"), nil, "need to set either File or ConfigMap"))
+	}
+	return allErrs
+}
+
+// ValidateKubeSchedulerLeaderElectionConfiguration ensures validation of the KubeSchedulerLeaderElectionConfiguration struct
+func ValidateKubeSchedulerLeaderElectionConfiguration(cc *componentconfig.KubeSchedulerLeaderElectionConfiguration, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	allErrs = append(allErrs, ValidateLeaderElectionConfiguration(&cc.LeaderElectionConfiguration, field.NewPath("leaderElectionConfiguration"))...)
+	if len(cc.LockObjectNamespace) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("lockObjectNamespace"), ""))
+	}
+	if len(cc.LockObjectName) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("lockObjectName"), ""))
+	}
+	return allErrs
+}
+
+// ValidateLeaderElectionConfiguration ensures validation of the LeaderElectionConfiguration struct
+func ValidateLeaderElectionConfiguration(cc *componentconfig.LeaderElectionConfiguration, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if cc.LeaseDuration.Duration <= 0 {
```

Comment:
I am not sure if 0 is a valid Duration. If it is not, the error message at the next line should be changed.

Reply:
I think no as well: https://github.com/kubernetes/kubernetes/pull/64218/files